File: python/tests/test_client.py
import unittest
from ttagy import TtagyClient, TtagyRequest, extract_structured_json

class TestTtagyClient(unittest.TestCase):
    def test_client_initialization(self):
        client = TtagyClient(
            base_url="http://127.0.0.1:8970",
            socket_path="/tmp/ttagy.sock",
            auth_token="secret",
            auto_fallback=True,
        )
        self.assertEqual(client.base_url, "http://127.0.0.1:8970")
        self.assertEqual(client.socket_path, "/tmp/ttagy.sock")
        self.assertEqual(client.auth_token, "secret")
        self.assertTrue(client.auto_fallback)

    def test_request_to_dict(self):
        req = TtagyRequest(
            prompt="测试",
            model="gemini-3.7-flash",
            effort="high",
            temperature=0.7,
        )
        d = req.to_dict()
        self.assertEqual(d["prompt"], "测试")
        self.assertEqual(d["model"], "gemini-3.7-flash")
        self.assertEqual(d["effort"], "high")
        self.assertEqual(d["temperature"], 0.7)

    def test_extract_structured_json(self):
        markdown = 'Here is the result:\n```json\n{\n  "status": "ok",\n  "count": 42\n}\n```\nDone!'
        extracted = extract_structured_json(markdown)
        self.assertIn('"status": "ok"', extracted)
        self.assertIn('"count": 42', extracted)

if __name__ == "__main__":
    unittest.main()

File: crates/ttagy-client/tests/test_client.rs
use ttagy_client::{ClientBuilder, TtagyRequest};

#[tokio::test]
async fn test_client_builder_and_fallback_config() {
    let client = ClientBuilder::default()
        .base_url("http://127.0.0.1:8970")
        .socket_path("/tmp/ttagy.sock")
        .auth_token("secret")
        .auto_fallback(true)
        .build()
        .expect("构建客户端成功");

    assert_eq!(client.config.base_url, Some("http://127.0.0.1:8970".to_string()));
    assert_eq!(client.config.auth_token, Some("secret".to_string()));
    assert!(client.config.auto_fallback);

    let req = TtagyRequest {
        prompt: "测试提示词".to_string(),
        model: Some("gemini-3.7-flash".to_string()),
        effort: Some("none".to_string()),
        ..Default::default()
    };

    assert_eq!(req.prompt, "测试提示词");
    assert_eq!(req.effort, Some("none".to_string()));
}
